[CS2103T-F13-2] $aveNUS

seanlowjk opened the pull request:
@jon-chua
@robytanama
@fatclarence
@Raikonen

kerwin97 commented on docs/DeveloperGuide.adoc:
```diff
@@ -361,14 +361,86 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+*MSS*
+
+1. User requests to list recommendations.
+2. $aveNUS shows a list of recommended food items based on their specified budget.
```
For this use case, could there be a  possible extension to choose the budget when searching for recommendations? A user might have to keep setting budget if he/she would like to increase the search. 
It is difficult to provide possible extensions as we are not sure how your program will be like. 
Use Cases are all satisfactory, fulfilling the tutorial's requirements. Step numbering correct,
possible extensions mentioned, no unnecessary UI details, no formatting/notional errors. Nicely done!

zhangxuan97 commented on docs/DeveloperGuide.adoc:
```diff
@@ -137,130 +146,245 @@ image:BetterModelClassDiagram.png[]
 
-===== Aspect: How undo & redo executes
+.Sequence diagram for sample save command
+image::SaveSequenceDiagram.png[]
```
Technical details:
- For this save sequence diagram, perhaps you can consider removing the participant SavingsAccount as it does not seem to be used. 
- For LogicManager, the activation bar seems to cut off prematurely - before Storage#saveSavingsAccount (it should deactivate at the bottom arrow instead).
- Storage should be StorageManager instead, as Storage is an interface.

Additionally, maybe you can consider splitting this sequence diagram (put a reference frame over LogicManager, Storage, and JsonStorageManager and include the storage components in the separate diagram).

zhangxuan97 reviewed (commented):
Generally, the developers guide is well written - our group feels that your recommend feature is very comprehensive. The sorting feature can be more elaborated on - perhaps you guys can make more reference to the class diagram that you drew.

TeoShyanJie commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +457,146 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`*` |forgetful user |save specific meal sets to reuse |save time on inputting my meals daily.
+|=======================================================================
 
```
Perhaps you could rearrange the user in alphabetical order to aid in understanding what a role can do.

seanlowjk committed:
Merge pull request #285 from jon-chua/modify-ppp

Modify PPP section

tanamaroby committed:
Remove unnecessary icons

tanamaroby committed:
Remove unnecessary imports

tanamaroby committed:
Fix color for light theme

seanlowjk committed:
Merge pull request #286 from robytanama/branch-Ui-changes

[v1.4] Very minor UI changes

fatclarence committed:
add withdraw capabilities without testing and without a savings account

fatclarence committed:
Merge pull request #284 from seanlowjk/update-sean-ppp

Update Sean's PPP

Raikonen committed:
Update UG and PPP

Raikonen committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-2/main

seanlowjk committed:
Merge pull request #288 from Raikonen/master

Update UG and PPP

Raikonen committed:
Merge branch 'branch-withdraw-2' of https://github.com/fatclarence/main into fatclarence-branch-withdraw-2

fatclarence committed:
add persistency for depositing money into savings account
add savings account
savings-account records savings but not withdrawals
update test cases and models

fatclarence committed:
complete withdraw feature with persistent saving into savings account
add addition and substraction functions within Money class for greater abstraction

fatclarence committed:
fix withdraw when savings account insufficient funds bug

fatclarence committed:
fix save command back to save

fatclarence committed:
fix withdraw bug

fatclarence committed:
update ppp for clarence

tanamaroby committed:
Draft PPP

tanamaroby committed:
Update Roby PPP

tanamaroby committed:
Merge pull request #289 from fatclarence/master

Update ppp for clarence

fatclarence committed:
Merge pull request #290 from robytanama/branch-PPP

[v1.3] Roby's PPP update

tanamaroby committed:
Add RepoSense

fatclarence committed:
Merge pull request #291 from robytanama/branch-PPP

[v1.4] Add RepoSense for Roby PPP

seanlowjk committed:
Rename PPPs

seanlowjk committed:
Merge branch 'master' into edit-adoc-names

fatclarence committed:
Merge pull request #293 from seanlowjk/edit-adoc-names

Edit Adoc Names

fatclarence committed:
Merge branch 'master' into branch-withdraw-2

fatclarence committed:
add colour to savings and withdrawals in css

fatclarence committed:
display savings history starting with most recent saving/withdrawal

fatclarence committed:
remove unused imports

Raikonen committed:
Update PPP

Raikonen committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-2/main into fatclarence-branch-withdraw-2

fatclarence committed:
add savings account display to the main window

fatclarence committed:
Merge branch 'branch-show-withdraw' into branch-withdraw-2

fatclarence committed:
final adjustments in ui

fatclarence committed:
add show command to restrict display for savingshistory

fatclarence committed:
fix UI problems for wallet and account panel

fatclarence committed:
Merge branch 'branch-show-withdraw' into branch-withdraw-2

fatclarence committed:
fix UI bugs for savings account

Raikonen committed:
Catch number format exception for daystoexpire

fatclarence committed:
add test for savecommand

seanlowjk committed:
Merge pull request #287 from fatclarence/branch-withdraw-2

Add Withdraw and Show Command and Savings Account

tanamaroby committed:
Place savings and budget into Vbox

tanamaroby committed:
Clean up MainWindow.java

tanamaroby committed:
Delete help icon

fatclarence committed:
fix limit on savings and budget

fatclarence committed:
increase savings limit to Integer.MAX

seanlowjk committed:
Merge pull request #297 from fatclarence/branch-withdraw-2

Fix bugs on Savings Account

jon-chua committed:
Fix recommendation system bugs and update UG/DG

jon-chua committed:
Further edits to User Guide

fatclarence committed:
add withdrawal info in infocommand

jon-chua committed:
Minor UGDG edits

fatclarence committed:
add show info in infocommand

seanlowjk committed:
Merge pull request #298 from jon-chua/branch-fix-recommendation-bugs

Improve recommendation system and updates to UG and DG

fatclarence committed:
update help window with show and withdraw

jon-chua committed:
Add MSS and test cases for recommendation system in DG

jon-chua committed:
Fix DG errors

fatclarence committed:
update user guide for save command

seanlowjk committed:
Merge pull request #299 from fatclarence/branch-info

Update InfoCommand with Show and Withdraw details

fatclarence committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-2/main

seanlowjk committed:
Edit Help and Info Window for Alias

tanamaroby committed:
Merge pull request #302 from seanlowjk/edit-alias-info

Edit Help and Info Window for Alias

tanamaroby committed:
Merge pull request #300 from jon-chua/master

Add MSS and manual testing for recommendation system

fatclarence committed:
update withdraw command in UG

fatclarence committed:
update UG for show command

Raikonen committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-2/main into branch-daystoexpire-hotfix

fatclarence committed:
update UG for history command

Raikonen committed:
Update Budget field constraints

Raikonen committed:
Sort puchase history by Time Of Purchase

fatclarence committed:
change current savings to be a private variable

fatclarence committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-2/main

tanamaroby committed:
Add first part of button sets

tanamaroby committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-2/main into branch-Ui-changes

Raikonen committed:
Snaeked

fatclarence committed:
Merge pull request #305 from Raikonen/branch-daystoexpire-hotfix

Budget Fixes

tanamaroby committed:
Finish all buttons

fatclarence committed:
Merge pull request #306 from Raikonen/branch-add-author

Add author tags

fatclarence committed:
update developer guide for savings feature

fatclarence committed:
update alias command in UG

tanamaroby committed:
Edit statistics UI

fatclarence committed:
update typo errors within the UG

tanamaroby committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-2/main into branch-Ui-changes

fatclarence committed:
add 'show' and 'withdraw' into the command summary

seanlowjk committed:
Add Author Annotations

fatclarence committed:
update UG command summary examples

Raikonen committed:
Merge pull request #301 from fatclarence/branch-update-ug

Update User Guide for Savings Account Feature

Raikonen committed:
Merge pull request #308 from seanlowjk/add-author-name

Add Author Annotations

seanlowjk committed:
Turn Recommendation System Off for Certain Commands

seanlowjk committed:
Update User Guide

seanlowjk committed:
Fix Recommend Auto-Sort Dependency

fatclarence committed:
update part of DG for savings feature

seanlowjk committed:
Edit Recommend Note

jon-chua committed:
Merge pull request #309 from seanlowjk/sorting-recommend-depndencies

Fix Sorting and Recommendation Dependencies

seanlowjk committed:
Fix Addition and Editing of Identical Foods

seanlowjk committed:
Fix equals and isSameFood

Raikonen committed:
Tag and Opening hours fixes

fatclarence committed:
Merge pull request #310 from seanlowjk/fix-duplicate-name-category

Fix Addition and Editing of Identical Foods

fatclarence committed:
edit developer guide for savings feature

fatclarence committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-2/main

fatclarence committed:
Merge branch 'master' into branch-update-dg

tanamaroby committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-2/main into branch-Ui-changes

Raikonen committed:
Add separate formatting and comparison check for opening hours

Raikonen committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-2/main

seanlowjk committed:
Add Food Component to DG

seanlowjk committed:
Merge pull request #311 from Raikonen/master

Tag and Opening hours fixes

seanlowjk committed:
Allow Upper and Lower Case for filter

seanlowjk committed:
Allow Upper and Lower Case for sorting functions

Raikonen committed:
Merge pull request #312 from seanlowjk/add-food-section-dg

Add Food Component to DG

fatclarence committed:
update the Making a saving section in  in DG

tanamaroby committed:
Add tooltip for buttons

tanamaroby committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-2/main into branch-Ui-changes

tanamaroby committed:
Merge pull request #313 from seanlowjk/change-sort-fields

Casing Changing for FieldParser and QuantifierParser

seanlowjk committed:
Edit Sorting Section for DG

jon-chua committed:
Added more design considerations to DG

seanlowjk committed:
Add Empty Line at EOF

seanlowjk committed:
Merge pull request #307 from robytanama/branch-Ui-changes

[v1.4] Changed the UI

tanamaroby committed:
Add Theme Command

jon-chua committed:
Edit recommend user guide photo

jon-chua committed:
Fix formatting

tanamaroby committed:
Add test for theme command

tanamaroby committed:
Fix checkstyle

tanamaroby committed:
Merge pull request #314 from seanlowjk/edit-sorting-dg

Edit Sorting Section for DG

tanamaroby committed:
Add info card for Theme command

tanamaroby committed:
Edit show and withdraw info cards

tanamaroby committed:
Fix alias info card not showing

tanamaroby committed:
Fix wrapping for info card

tanamaroby committed:
Merge pull request #316 from jon-chua/branch-edit-dg-2

Edit developer guide of recommendation system

tanamaroby committed:
Update UG on Theme

tanamaroby committed:
Update UG

tanamaroby committed:
Edit Food Card

jon-chua committed:
Merge pull request #315 from robytanama/branch-Ui-changes

[v1.4] Add theme command

tanamaroby committed:
Fix tooltip bug

fatclarence committed:
update add savings implementation

jon-chua committed:
Merge pull request #317 from robytanama/branch-Ui-changes

[v1.4] Fix Tooltip bug

fatclarence committed:
update withdrawal implementation details

fatclarence committed:
update show in savings implementation

seanlowjk committed:
Make UG Changes

seanlowjk committed:
Fix Command Summary

seanlowjk committed:
Make Food Item Commands Clearer

seanlowjk committed:
Fix Price In User Guide

tanamaroby committed:
Merge pull request #318 from seanlowjk/major-ug-dg-updates

More UG Updates

seanlowjk committed:
Fix Diagram Typo

fatclarence committed:
add class diagrams for savings feature

fatclarence committed:
update EOF in savings history class diagram

seanlowjk committed:
Edit About Us for Sean and Clarence

fatclarence committed:
add minor updates for SaveSequenceDiagram

fatclarence committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-2/main

tanamaroby committed:
First part of UG info command

tanamaroby committed:
Finish Info feature of Dev Guide

tanamaroby committed:
Merge pull request #319 from seanlowjk/fix-typos

Fix Diagram Typo

tanamaroby committed:
Merge pull request #321 from seanlowjk/edit-aboutus

Edit About Us for Sean and Clarence

tanamaroby committed:
Merge pull request #320 from fatclarence/branch-update-dg

Update Savings Feature in Developer Guide

fatclarence committed:
add test for savings history storage

fatclarence committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-2/main into branch-save-test

tanamaroby committed:
Merge pull request #322 from fatclarence/branch-save-test

Add test for SavingsStorage and SavingsHistory

tanamaroby committed:
Finish design consideration

fatclarence committed:
add test for JsonAdaptedSavings

seanlowjk committed:
Merge pull request #324 from fatclarence/branch-save-test

Add test coverage for JsonAdaptedSavings

tanamaroby committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-2/main into branch-PPP

tanamaroby committed:
Remove class diagram

tanamaroby committed:
Edit numbering format

fatclarence committed:
Merge pull request #323 from robytanama/branch-PPP

[v1.4] Update DG for Info feature

Raikonen committed:
Minor UG changes

fatclarence committed:
add SaveSequenceDiagram

tanamaroby committed:
Merge pull request #325 from Raikonen/branch-ug-changes

Minor UG changes

fatclarence committed:
update SaveActivityDiagram in DG

jon-chua committed:
Edit UG and DG

seanlowjk committed:
Minor UG Updates

seanlowjk committed:
Merge pull request #327 from jon-chua/edit-dg

Minor edits to DG

seanlowjk committed:
Fix equals() method for Food

seanlowjk committed:
Edit Appendix F of DG

seanlowjk committed:
Add Use Cases

fatclarence committed:
Merge pull request #330 from seanlowjk/edit-appendix-f

Add Use and Test Cases

fatclarence committed:
Merge pull request #328 from seanlowjk/fix-ug-clarity

Minor UG Updates

seanlowjk committed:
Fix License URL

fatclarence committed:
Merge pull request #329 from seanlowjk/fix-food-bug

Fix equals() method for Food

seanlowjk committed:
Merge pull request #326 from fatclarence/branch-update-dg

Update the Developer Guide

seanlowjk committed:
Make Changes to Sean's PPP

tanamaroby committed:
Update DG on theme feature

tanamaroby committed:
Accommodate theme in alias

seanlowjk committed:
Remove Alias Section

tanamaroby committed:
Updated UG

tanamaroby committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-2/main

tanamaroby committed:
Fix test

fatclarence committed:
update clarence ppp

seanlowjk committed:
Merge pull request #333 from robytanama/master

[v1.4] Add Theme feature into DG

tanamaroby committed:
Merge pull request #332 from seanlowjk/sean-ppp-changes

Make Changes to Sean's PPP

tanamaroby committed:
Merge pull request #331 from seanlowjk/fix-license

Fix License URL

fatclarence committed:
resolve merge conflicts

fatclarence committed:
edit itallics in contributions to user guide

tanamaroby committed:
Shortened Info in DG

tanamaroby committed:
Merge pull request #334 from fatclarence/branch-update-ppp

Update PPP for Clarence

tanamaroby committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-2/main

seanlowjk committed:
Fix URL Printing

fatclarence committed:
update v2.0 commands in UG

tanamaroby committed:
Update PPP for Roby

tanamaroby committed:
Merge pull request #336 from seanlowjk/edit-gh-pages-css

Fix URL Printing

seanlowjk committed:
Merge pull request #335 from robytanama/master

[v1.4] Update Roby's PPP

seanlowjk committed:
Add Page Breaks

fatclarence committed:
Merge pull request #337 from seanlowjk/finalize-sean-ppp

Add Page Breaks

tanamaroby committed:
Update robytanama.adoc

tanamaroby committed:
Update robytanama.adoc

tanamaroby committed:
Update robytanama.adoc

tanamaroby committed:
Update robytanama.adoc

jon-chua committed:
Merge pull request #338 from AY1920S1-CS2103T-F13-2/robytanama-patch-1

[v.1.4] Update robytanama.adoc with Page Break

tanamaroby committed:
Fix Page Break

seanlowjk committed:
Add more page breaks

tanamaroby committed:
Merge pull request #339 from seanlowjk/finalize-sean-ppp

Add more page breaks

Raikonen committed:
Update UI Images

fatclarence committed:
update UI for savings in UG

seanlowjk committed:
Update seanlowjk.adoc

Raikonen committed:
Merge pull request #341 from fatclarence/branch-update-ppp

Update UI in savings feature in UG

fatclarence committed:
Merge pull request #340 from Raikonen/branch-ug-changes

Update UI Images

tanamaroby committed:
Remove Page break

tanamaroby committed:
Update Roby's PPP

seanlowjk committed:
Merge pull request #342 from fatclarence/branch-update-ppp

Update UI for savings in UG

fatclarence committed:
update fatclarence ppp

fatclarence committed:
align images in userguide to center

fatclarence committed:
Merge branch 'branch-update-ppp' into branch-update-ug

seanlowjk committed:
Finalize Sorting Section Tests

seanlowjk committed:
Merge pull request #343 from fatclarence/branch-update-ppp

Update PPP for Clarence

jon-chua committed:
Remove some sections from PPP

seanlowjk committed:
Merge pull request #345 from jon-chua/edit-ppp

Edit Jon's PPP

seanlowjk committed:
Align Images to the Center

Raikonen committed:
Add design considerations for purchase model

Raikonen committed:
Merge pull request #346 from seanlowjk/finalize-developer-guide

Align Images to the Center for Developer Guide

Raikonen committed:
Merge pull request #344 from seanlowjk/add-sort-tests

Finalize Sorting Section Tests

jon-chua committed:
Merge pull request #347 from Raikonen/branch-dg-changes

Add design considerations for purchase model

fatclarence committed:
update ug

Raikonen committed:
Resize images

Raikonen committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-2/main into branch-ug-changes

fatclarence committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-2/main into branch-update-ug

Raikonen committed:
Merge pull request #349 from fatclarence/branch-update-ug

Update UG images align center

Raikonen committed:
Fix merge conflicts

seanlowjk committed:
Merge pull request #348 from Raikonen/branch-ug-changes

Branch ug changes

jon-chua committed:
Add captions for Recommendation System UG

Raikonen committed:
Add more stuff to PPP

seanlowjk committed:
Merge pull request #350 from jon-chua/master

Add captions for User Guide recommendation system

Raikonen committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-2/main into branch-dg-changes

Raikonen committed:
Merge pull request #351 from Raikonen/branch-dg-changes

Add more stuff to PPP

fatclarence committed:
fix ui.png
update test for savecommandparser

fatclarence committed:
fix checks

Raikonen committed:
Merge pull request #352 from fatclarence/branch-update-ug

Fix DarkMode.png in UserGuide

fatclarence committed:
add test for withdraw command

seanlowjk committed:
Merge pull request #353 from fatclarence/branch-update-ug

Add test for withdraw command

fatclarence committed:
add savings test for savingshistory

fatclarence committed:
add author

seanlowjk committed:
Merge pull request #354 from fatclarence/branch-update-ug

Add savings test for SavingsHistory

tanamaroby committed:
Fix viewsort info bug

seanlowjk committed:
Merge pull request #355 from robytanama/branch-PPP

[v1.4] Fix viewsort info bug
